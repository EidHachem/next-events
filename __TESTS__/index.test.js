import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import HomePage from "../pages/index"
import { useRouter } from "next/router"

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}))

describe("Home", () => {
  it("renders the home page when the user visits the website without crashing", () => {
    useRouter.mockImplementation(() => ({
      route: "/",
      pathname: "/",
      query: {},
      asPath: "/",
    }))
    render(<HomePage />)
    const { container } = render(<HomePage />)
    expect(container).toBeInTheDocument()
  })

  it("renders a list", () => {
    render(<HomePage />)

    const list = screen.getByRole("list")

    expect(list).toBeInTheDocument()
  })

  it("renders homepage unchanged", () => {
    const { container } = render(<HomePage />)
    expect(container).toMatchSnapshot()
  })
})
