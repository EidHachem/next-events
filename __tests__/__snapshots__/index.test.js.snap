// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <img
          alt="Networking for introverts"
          src="/images/introvert-event.jpg"
        />
        <div
          class="content"
        >
          <div
            class="summary"
          >
            <h2>
              Networking for introverts
            </h2>
            <div
              class="date"
            >
              <time>
                May 30, 2021
              </time>
            </div>
            <div
              class="address"
            >
              <address>
                New Wall Street 5
98765 New Work
              </address>
            </div>
          </div>
          <div
            class="actions"
          >
            <a
              class="btn"
              href="/events/e2"
            >
              Explore Event
            </a>
          </div>
        </div>
      </li>
      <li
        class="item"
      >
        <img
          alt="Networking for extroverts"
          src="/images/extrovert-event.jpg"
        />
        <div
          class="content"
        >
          <div
            class="summary"
          >
            <h2>
              Networking for extroverts
            </h2>
            <div
              class="date"
            >
              <time>
                April 10, 2022
              </time>
            </div>
            <div
              class="address"
            >
              <address>
                My Street 12
10115 Broke City
              </address>
            </div>
          </div>
          <div
            class="actions"
          >
            <a
              class="btn"
              href="/events/e3"
            >
              Explore Event
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
